hal/spi_async_api.h

```diff
@@ -0,0 +1,178 @@
+/** \ingroup hal_spi */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_SPI_API_ASYNC_H
+#define MBED_SPI_API_ASYNC_H
+
+#include "hal/spi_api.h"
+
+#if DEVICE_SPI
+#if DEVICE_SPI_ASYNCH
```

Comment:
I think `DEVICE_SPI_ASYNCH` has so far only been implemented on a very few targets. (I think I heard one vendor implemented it for their own product, but others didn't follow.) Should it remain as an optionally-implemented feature, or will it be required for all SPI implementations? If required, the `#define` should go away, right? If optional, will we see more targets implementing it?

Reply:
Actually a lot of target do implement it.
All the stm32 family, and a lot of other vendors including SiLabs, Atmel, Nordic etc.
It is still optional as some device cannot implement it though.